### PR TITLE
Initial version of deprecation process

### PR DESCRIPTION
**Pull request by benjagm**
This is the Initial version of a deprecation process to better handle deprecations.

**Summary**: In #493 we agreed on preparing a deprecation process to communicate the deprecation of the http://json-schema.org/schema redirect. This is a first proposal to have an standard process for cases like this.

**Do you think resolving this issue might require an [Architectural Decision Record (ADR)](https://github.com/json-schema-org/community/blob/main/CONTRIBUTING.md#key-architectural-decisions)? (significant or noteworthy)**

No

### CODE REVIEW

File: docs/deprecation-plan.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# The process of deprecating a feature
+
+Once the decision to deprecate a feature has been made through an ADR , below is a recommend set of actions:
```
Comment (jdesrosiers) [inline docs/deprecation-plan.md:3]:
```suggestion
Once the decision to deprecate a feature has been made through an ADR, below is a recommend set of actions:
```

File: docs/deprecation-plan.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+1. **Determine if you need an exception process**. Some features may require extended migration periods to allow customers extended time to fully adopt a new solution.
+
+2. **Craft the message & build the internal enablement document**. This document should include frequently asked questions, timelines, and any other relevant information. The best approach here is to publish a blog post.
```
Comment (jdesrosiers) [inline docs/deprecation-plan.md:7]:
```suggestion
2. **Craft the message and build the internal enablement document**. This document should include frequently asked questions, timelines, and any other relevant information. The best approach here is to publish a blog post.
```

**Review by jdesrosiers — COMMENTED** [review]
What's the scope of this document? I know the inspiration for it is no longer serving `http://json-schema.org/latest`, but is this intended to also refer to spec changes? For example, deprecating a keyword? I think that's a different context that would require some different considerations including a much longer deprecation period before removal.

**Comment by benjagm** [comment]
> What's the scope of this document?

AS agreed in the last OCWM #515 I'll add a scope section to make it clear that this is not applicable to the spec development.

**Review by jdesrosiers — APPROVED** [review]

**Comment by jdesrosiers** [comment]
I think my suggestion at the OCWM was misunderstood. I wasn't concerned about non-breaking changes. I'm fine with including that, but that wasn't the point. What I was trying to communicate was that we would want to follow this plan for any breaking changes. Deprecation is just one kind of breaking change. I think it would be wise to change some of the wording in this document so that it generalizes to any breaking change, not just deprecation.

**Comment by benjagm** [comment]
> I think it would be wise to change some of the wording

You are right @jdesrosiers. I think you didn't said anything about breaking or non-breaking changes, it is just I choose the wrong words to differentiate the new section with the recommendations.

**Comment by benjagm** [comment]
@jdesrosiers I pushed some changes to improve the wording as we discussed in the last OCWM. Please let me know what do you think about this new version.

File: docs/deprecation-plan.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  - When is this change comming into effect? The deprecation timeline.
+
+The announcement should communicate the timeline and the related ADR. This will be shared before sharing it publicly with users (This can be done in the same day).
+
+4. **Share with users at minimum 30-days in advance**. It is recommended to start with a broad social media announcement including a reference to a blog post with the same content than the Github announcement. 
+
+
+5. **Remind users twice**. 7 days and 1 day before. This is good practice to ensure that users hear the message and are following any migration steps required. The announcement will be shared in social media and Slack #announcements.
+
+6. **Execute the deprecation**.
```
Comment (jdesrosiers) [inline docs/deprecation-plan.md:26]:
```suggestion
6. **Execute the change**.
```

**Review by jdesrosiers — APPROVED** [review]
Looks good! This is the change I was looking for. There's one place where we probably want to say "change" instead "deprecation" I pointed out inline and we might want to change the filename for the document as well to say "change" instead of "deprecation".

**Pull request merged**